pkg/operator/status/inertia.go

```diff
@@ -0,0 +1,56 @@
+package status
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+)
+
+// Inertia returns the inertial duration for the given condition.
+type Inertia func(condition operatorv1.OperatorCondition) time.Duration
+
+// InertiaCondition configures an inertia duration for a given set of
+// condition types.
+type InertiaCondition struct {
+	// ConditionTypeMatcher is a regular expression selecting condition types
+	// with which this InertiaCondition is associated.
+	ConditionTypeMatcher *regexp.Regexp
+
+	// Duration is the inertial duration for associated conditions.
+	Duration time.Duration
+}
+
+// InertiaConfig holds configuration for an Inertia implementation.
+type InertiaConfig struct {
+	defaultDuration time.Duration
+	conditions      []InertiaCondition
+}
+
+// NewInertia creates a new InertiaConfig object.  Conditions are
+// applied in the given order, so a condition type matching multiple
+// regular expressions will have the duration associated with the first
+// matching entry.
+func NewInertia(defaultDuration time.Duration, conditions ...InertiaCondition) (*InertiaConfig, error) {
```

Comment:
make a `MustNewIntertia` that panics on the error to ease the coding.

Reply:
> make a `MustNewIntertia` that panics on the error to ease the coding

Done with cb0967e89 -> 2d2aa9635, which also rebases onto the current master (no conflicts).